pilot/pkg/xds/debug.go

```diff
@@ -448,18 +444,12 @@ func (s *DiscoveryServer) adsz(w http.ResponseWriter, req *http.Request) {
 	w.Header().Add("Content-Type", "application/json")
 	if req.Form.Get("push") != "" {
 		AdsPushAll(s)
-		s.adsClientsMutex.RLock()
-		_, _ = fmt.Fprintf(w, "Pushed to %d servers", len(s.adsClients))
-		s.adsClientsMutex.RUnlock()
+		_, _ = fmt.Fprintf(w, "Pushed to %d servers", len(s.Clients()))
```

Comment:
doing `s.adsClientCount()` here would avoid allocation in `Clients()` I think? (not sure it matters in practice)

Reply:
Probably doesn't matter since its a debug endpoint, but might as well make it more optimized. thanks